Sources/StreamChatUI/ChatMessageList/ChatTypingIndicatorView.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+// TODO: Decide which IndicatorView to Use:
+// A) AChatTypingIndicatorView
+// B) ChatTypingIndicatorView
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+public typealias AChatTypingIndicatorView = _ChatTypingIndicatorView<NoExtraData>
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+public typealias ChatTypingIndicatorView = _ChatTypingIndicatorView<NoExtraData>
+
+/// A `UIView` subclass with 3 dots which can be animated with fading out effect.
+open class _ChatTypingIndicatorView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
+    open var dotSize: CGSize = .init(width: 5, height: 5)
+
+    private(set) lazy var innerView: UIView = UIView().withoutAutoresizingMaskConstraints
+
+    private var viewWidthConstant: CGFloat {
+        dotLayer.frame.size.width * 1.5 * 3 + directionalLayoutMargins.leading * 2
+    }
+
+    open private(set) lazy var dotLayer: CALayer = {
+        let layer = CALayer()
+
+        layer.frame.size = dotSize
+
+        layer.backgroundColor = UIColor.red.cgColor
+        layer.cornerRadius = dotSize.height / 2
+        return layer
+    }()
+
+    open private(set) lazy var replicatorLayer = CAReplicatorLayer()
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+        embed(innerView, insets: directionalLayoutMargins)
```

Comment:
We need to remove this API 😱 It's wrong. You want to do `innerView.pin(to: layoutMarginsGuide)`